### PR TITLE
Link the Symbol of Constants Used as the Array Length

### LINKED ISSUE #31184: Symbol and References APIs are not working for constant as the array length
**Description:**
Consider the following source snippet

```
import ballerina/module1;

public type TEST_TYPE int[CON<cursor>ST1];

function testFunction() {
}

public const CONST1 = 12;
```

Invoking the symbol API at the given cursor position is not providing the actual constant symbol

**Affected Versions:**
SwanLake-beta1 at least

### PR DESCRIPTION
**Pull request by sanjana**
Fixes #31184

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java [hunk 1]
```diff
@@ -1051,6 +1051,7 @@ public void visit(BLangArrayType arrayTypeNode) {
                     Name typeName = names.fromIdNode(sizeReference.variableName);
 
                     BSymbol sizeSymbol = lookupMainSpaceSymbolInPackage(size.pos, env, pkgAlias, typeName);
+                    ((BLangSimpleVarRef) size).symbol = sizeSymbol;
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java:1054]: Can use `sizeReference` here.

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java [hunk 2]
```diff
@@ -1133,6 +1133,9 @@ public void visit(BLangValueType valueType) {
     @Override
     public void visit(BLangArrayType arrayType) {
         lookupNode(arrayType.elemtype);
+        for (BLangExpression size : arrayType.sizes) {
+            lookupNode(size);
+        }
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java:1138]: Can use `lookupNodes(arrayType.sizes)`

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java (hunk 1 above)
Comment (sanjana) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java]: Fixed.

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java (hunk 2 above)
Comment (sanjana) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java]: Please note that the `arrayType.sizes` is of type `BLangExpression[]`.
The `lookupNodes()` accepts a parameter in a type of `List<? extends BLangNode> nodes`.  Therefore, a type casting was performed using `Arrays.asList()`.

**Review by pubudu91 — APPROVED** [review]

**Pull request merged**